### ISSUE TITLE
Cannot depend on a git package if the git package depends on a path package.

### ISSUE THREAD

**Issue by xuan-nguyen-swe** [dart-lang/pub#2447]
I have a problem when I want to add dependency from another team.

Assume my company has a partner. The partner allows my company access to their git repository `ssh://git@partner.com/root_project.git`. In `root_project`, we have 2 sub-projects: `project_a` and `project_b`. Below is the `pubspec.yaml` of `project_a`:

```
# /project_a/pubspec.yaml

dependencies:
  project_b:
    path: ../project_b
```

My `pubspec.yaml`:
```
dependencies:
  project_a:
    git:
      url: ssh://git@partner.com/root_project.git
      path: project_a
```

If I run `pub get`, I hope that it's smart enough to find `project_b` from relative path of the git repository.

Unfortunately, I get error: `dependency: "../project_b" is a relative path, but this isn't a local pubspec.`

The problem is that I have no permission to change the file `/project_a/pubspec.yaml` to use git url because it's not my project.

**Comment by sigurdm** [comment]
Yes, my guess is that these are the same.

Is the path being interpreted relative to the git root?
Ie. what happens if you use:

```
# /project_a/pubspec.yaml

dependencies:
  project_b:
    path: project_b
```

**Comment by nguyenxndaidev** [comment]
> Is the path being interpreted relative to the git root?
* The partner project has 2 sub projects (project_a depends on project_b). Both are belong to the partner.
* project_a point to project_b using local path
* The configuration of local path is correct and relative to the repository root.

> Ie. what happens if you use:
> 
> ```
> # /project_a/pubspec.yaml
> 
> dependencies:
>   project_b:
>     path: project_b
> ```
As I mentioned, project_a is not my project so I cannot change `/project_a/pubspec.yaml`.

My project depends on project_a:
```
dependencies:
  project_a:
    git:
      url: ssh://git@partner.com/root_project.git
      path: project_a
```

**Comment by sigurdm** [comment]
It was not meant as a solution, just to explore what is currently happening when you have a path dependency in a git dependency.

**Comment by sigurdm** [comment]
Nevermind - I realize now that pub doesn't support relative paths from git at all.

**Comment by jonasfj** [comment]
You can work around this with `dependency_overrides` (this is not ideal). You can also make a local clone of `root_project` and use path-dependencies in your project. I would suggest avoid use of path-dependencies in `root_project`, such that it can be depended on directly from a `pubspec.yaml`.

Another option for sharing private dependencies is to use a [private pub repository](https://blog.cloudsmith.io/2020/02/03/worlds-first-private-dart-repository-service-w-cloudsmith/).

----
I think the fix for `pub` is https://github.com/dart-lang/pub/issues/449, and that we can close this as a duplicate. I don't think this is a priority for us right now, if someone wants to work on and has an idea for how to solve it feel free to ping me on the original issue.

**Comment by nguyenxndaidev** [comment]
> You can work around this with `dependency_overrides` (this is not ideal)

Thanks for suggestion! Currently I checkout project locally as workaround. But `dependency_overrides` seems to be easier.

**Comment by nguyenxndaidev** [comment]
> You can work around this with `dependency_overrides` (this is not ideal)

Tested but unfortunately it doesn't help.

**Comment by zmtzawqlp** [comment]
any update?  dependency_overrides  is not work at my side.

**Comment by sigurdm** [comment]
I don't think we'll prioritize work on this issue in the short-term future.

Your best bet might be to make a local check-out and use a path-dependency to that.

**Comment by jonasfj** [comment]
Reading more carefully, I'm not sure this is a dup of https://github.com/dart-lang/pub/issues/449

**Comment by chengyuhui** [comment]
`dependency_overrides` seems to be not effective since the error here looks like being thrown before the overrides are even considered?

**Comment by spydon** [comment]
That `dependency_overrides` doesn't work in this specific case makes it very hard to work with non-published versions of packages on monorepos that use path dependencies, like the structure we have in [Flame](https://github.com/flame-engine/flame/tree/main/packages). This should be labeled as a bug imho.

**Comment by spydon** [comment]
Dived into the code a bit, the problem is that the `description` is parsed and checked before the `dependencyOverrides` are checked [here](https://github.com/spydon/pub/blob/master/lib/src/pubspec.dart#L483-L500).
I don't know much about pub, but maybe that check could just be skipped if `_dependencyOverrides` contains the currently checked package name?

**Comment by Cloudef** [comment]
I'm currently using path to workaround https://github.com/flutter/flutter/issues/63240 but this behavior certainly isn't helpful as now the dependency can't be used from git.

**Comment by sadi304** [comment]
Anyone found any work around for this case?

**Comment by ankit1kmr** [comment]
any update on this other than cloning the dependency locally and giving the path ?

**Comment by sigurdm** [comment]
Closing this as duplicate of https://github.com/dart-lang/pub/issues/449

**Comment by MohammedSalehelShazly** [comment]
see [this](https://github.com/felangel/bloc/issues/151#issuecomment-1385978523) solution it may help you.

**Comment by RubenGarcia** [comment]
I am also affected

**Comment by sigurdm** [comment]
@RubenGarcia this should be fixed in dart 3.4 and onwards: https://github.com/dart-lang/sdk/blob/main/CHANGELOG.md#pub-2
